From 13e9d32ce695af9752a4597ae5ffca6baa9c8675 Mon Sep 17 00:00:00 2001
From: Daniel Vetter <daniel.vetter@ffwll.ch>
Date: Wed, 10 Sep 2014 12:43:54 +0200
Subject: [PATCH 1287/1369] drm: unexport drm_global_mutex

commit 44af3f5c6a2dc0358121b5ecf272361b3290649e upstream.

Drivers really, really have no business even looking at this lock. And
thankfully they don't.

So unexport it and move the declaration to drm_internal.h.

Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Pengyu Ma <pengyu.ma@windriver.com>
---
 drivers/gpu/drm/drm_fops.c     |    1 -
 drivers/gpu/drm/drm_internal.h |    1 +
 drivers/gpu/drm/drm_lock.c     |    1 +
 include/drm/drmP.h             |    1 -
 4 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/drm_fops.c b/drivers/gpu/drm/drm_fops.c
index 51a77a7..3e66946 100644
--- a/drivers/gpu/drm/drm_fops.c
+++ b/drivers/gpu/drm/drm_fops.c
@@ -43,7 +43,6 @@
 
 /* from BKL pushdown */
 DEFINE_MUTEX(drm_global_mutex);
-EXPORT_SYMBOL(drm_global_mutex);
 
 static int drm_open_helper(struct file *filp, struct drm_minor *minor);
 
diff --git a/drivers/gpu/drm/drm_internal.h b/drivers/gpu/drm/drm_internal.h
index a71cf4b..d6dc755 100644
--- a/drivers/gpu/drm/drm_internal.h
+++ b/drivers/gpu/drm/drm_internal.h
@@ -25,6 +25,7 @@
 extern unsigned int drm_timestamp_monotonic;
 
 /* drm_fops.c */
+extern struct mutex drm_global_mutex;
 int drm_lastclose(struct drm_device *dev);
 
 /* drm_pci.c */
diff --git a/drivers/gpu/drm/drm_lock.c b/drivers/gpu/drm/drm_lock.c
index 727b032..f861361 100644
--- a/drivers/gpu/drm/drm_lock.c
+++ b/drivers/gpu/drm/drm_lock.c
@@ -36,6 +36,7 @@
 #include <linux/export.h>
 #include <drm/drmP.h>
 #include "drm_legacy.h"
+#include "drm_internal.h"
 
 static int drm_notifier(void *priv);
 
diff --git a/include/drm/drmP.h b/include/drm/drmP.h
index 187a9a3..290ec09 100644
--- a/include/drm/drmP.h
+++ b/include/drm/drmP.h
@@ -1084,7 +1084,6 @@ extern long drm_compat_ioctl(struct file *filp,
 extern bool drm_ioctl_flags(unsigned int nr, unsigned int *flags);
 
 				/* Device support (drm_fops.h) */
-extern struct mutex drm_global_mutex;
 extern int drm_open(struct inode *inode, struct file *filp);
 extern ssize_t drm_read(struct file *filp, char __user *buffer,
 			size_t count, loff_t *offset);
-- 
1.7.5.4

